### PR TITLE
eager quant: convert mapping for fused QAT Linear-Bn1d

### LINKED ISSUE #66483: Quantization - Linear-Bn1d fusion should support QAT
## 🐛 Bug

https://github.com/pytorch/pytorch/pull/50748 added Linear-Bn1d fusion for PTQ.  We should also have it for QAT.  This is useful for vision transformers.

high level steps to enable this:
1. add a nni.LinearBn1d module
2. add a nniqat.LinearBn1d module
3. create the mapping and the logic to swap from (1) to (2)
4. create the mapping and the logic from (2) to quantized.Linear module
5. test everything in eager and FX

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72796
* #72431

Summary:

Adds the eager mode convert mapping for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34213150](https://our.internmc.facebook.com/intern/diff/D34213150)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).

File: test/quantization/eager/test_quantize_eager_qat.py [hunk 1]
```diff
@@ -1006,6 +1006,22 @@ def test_linear_bn_numerics(self):
         r2 = m(data)
         self.assertTrue(torch.allclose(r1, r2))
 
+    @override_qengines
+    def test_linear_bn_workflow(self):
+        qengine = torch.backends.quantized.engine
+        m = nn.Sequential(
+            QuantStub(),
+            nn.Linear(4, 4),
+            nn.BatchNorm1d(4),
```
Comment (jerryzh168) [inline test/quantization/eager/test_quantize_eager_qat.py:1015]: optional nit: maybe add a DeQuantStub, just for completeness..

**Review by jerryzh168 — APPROVED** [review]

**Comment by jerryzh168** [comment]
looks like there are some errors from numeric suite

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).

**Pull request closed without merging**